### PR TITLE
Move Dockerfile from root to test/e2e/coexistence

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

- Move Dockerfile from root to test/e2e/coexistence
 
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- Run coexistence tests on local

```
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (coexistence/move-dockerfile)> ./hack/scripts/process-ginkgo-test-results.sh test/e2e/coexistence/testresults
| :memo: Test Suite Description | Total Tests | Passed | Failed | Skipped |
| --- | ---: | ---: | ---: | ---: |
| CLI Coexistence E2E Test Suite | 9 | 9 | 0 | 0 |
| **Total** | **9** | **9** | **0** | **0** |
mpanchajanya@mpanchajanF09MK ~/v/r/tanzu-cli (coexistence/move-dockerfile)> 


```

### Release note

```release-note

```

### CODE REVIEW

File: .github/workflows/cli-coexistence_tests.yaml [hunk 1]
```diff
@@ -17,24 +17,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v4
```
Comment (marckhouzam) [inline .github/workflows/cli-coexistence_tests.yaml:20]: Why this change?
Our other workflows are still on @v1
Should we be migrating all of them?

File: .github/workflows/cli-coexistence_tests.yaml [hunk 2]
```diff
@@ -17,24 +17,14 @@ jobs:
...
 
       - name: Set up Go 1.x
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v4
```
Comment (marckhouzam) [inline .github/workflows/cli-coexistence_tests.yaml:23]: Why this change?

File: .github/workflows/cli-coexistence_tests.yaml [hunk 3]
```diff
@@ -17,24 +17,14 @@ jobs:
...
         with:
           go-version: 1.21
         id: go
 
-      - name: go cache
```
Comment (marckhouzam) [inline .github/workflows/cli-coexistence_tests.yaml:28]: Why this change?

File: .github/workflows/cli-coexistence_tests.yaml [hunk 4]
```diff
@@ -50,10 +40,10 @@ jobs:
           make start-test-central-repo
 
       - name: Build Docker image
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v5
```
Comment (marckhouzam) [inline .github/workflows/cli-coexistence_tests.yaml:43]: Why this change?

File: .github/workflows/cli-coexistence_tests.yaml [hunk 5]
```diff
@@ -67,5 +57,11 @@ jobs:
       - name: Tests Results Summary
         if: always()
         run: |
-          TEST_RESULTS_MD=$(./hack/scripts/process-ginkgo-test-results.sh test/e2e/coexistence)
+          if [ -d "test/e2e/coexistence/testresults" ]; then
+          echo "Coexistence Test Results exists"
+          else
+            echo "Coexistence Test Results does not exist"
+          fi
```
Comment (marckhouzam) [inline .github/workflows/cli-coexistence_tests.yaml:64]: Is the above for debugging or is it still valid?
If valid, the indentation is wrong.

**Review by marckhouzam — APPROVED** [review]
Except for the github action changes that I don't understand, this LGTM
Thanks for cleaning this up!

File: .github/workflows/cli-coexistence_tests.yaml (hunk 5 above)
Comment (mpanchajanya) [reply .github/workflows/cli-coexistence_tests.yaml:64]: yes This is for testing purpose. removed the logs

File: .github/workflows/cli-coexistence_tests.yaml (hunk 2 above)
Comment (mpanchajanya) [reply .github/workflows/cli-coexistence_tests.yaml:23]: Updated to latest versions since few actions were deprecated

File: .github/workflows/cli-coexistence_tests.yaml (hunk 1 above)
Comment (mpanchajanya) [reply .github/workflows/cli-coexistence_tests.yaml:20]: Updated to latest versions since few actions were deprecated .

Yes we should migrate other workflows as well. I will file a JIRA.

File: test/e2e/coexistence/Dockerfile [hunk 6]
```diff
@@ -21,12 +21,12 @@ ENV TANZU_CLI_BUILD_VERSION=$TANZU_CLI_BUILD_VERSION \
     TANZU_CLI_COEXISTENCE_NEW_TANZU_CLI_DIR=$TANZU_CLI_COEXISTENCE_NEW_TANZU_CLI_DIR \
     TANZU_CLI_COEXISTENCE_LEGACY_TANZU_CLI_VERSION=$TANZU_CLI_COEXISTENCE_LEGACY_TANZU_CLI_VERSION
 
-# Install dependencies and go1.19
+# Install dependencies and go1.21
```
[the remark this answers is not in the document]
Comment (mpanchajanya) [reply test/e2e/coexistence/Dockerfile]: Thanks Updated the docker image to golang

File: .github/workflows/cli-coexistence_tests.yaml (hunk 1 above)
Comment (marckhouzam) [reply .github/workflows/cli-coexistence_tests.yaml:20]: Thanks!
